checker/jdk/nullness/src/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -4519,7 +4519,7 @@ public final String toString() {
             return sb.append(']').toString();
         }
 
-        public final boolean containsAll(Collection<?> c) {
+        public final boolean containsAll(Collection<? extends @NonNull Object> c) {
```

Comment:
It looks like these methods probably accept `null` elements in practice.

The only documentation of this fact is that they inherit the doc from `Collection.containsAll`, etc. and don't request to _also_ inherit the `@throws` doc.

That could easily be accidental, especially because this class didn't use to be publicly accessible. Well, _this class_ still isn't publicly accessible, but its subclass `ConcurrentHashMap.KeySetView` is as of 1.8 after being hidden before.

It's also a bit surprising that the collection views' query methods permit null elements when methods on `ConcurrentHashMap` itself do not.

I'm not sure what you want to do with all that :) But that's my read of the situation.

Reply:
As with the other example, I won't take (possibly accidental) omission of `@throws` to mean that null is a legal argument.